src/core/text/TextStyle.js

```diff
@@ -767,16 +758,16 @@ function areArraysEqual(array1, array2)
 /**
  * Utility function to ensure that object properties are copied by value, and not by reference
  *
- * @param {Object} target Target object to copy property into
- * @param {Object} source Source object for the proporty to copy
- * @param {string} prop Property name to copy
+ * @param {Object} target Target object to copy properties into
+ * @param {Object} source Source object for the proporties to copy
+ * @param {string} propertyObj Object containing properties names we want to loop over
  */
-function deepCopyProperty(target, source, prop) {
-    if (Array.isArray(source[prop])) {
-        target[prop] = source[prop].slice();
-    } else if (source[prop] && typeof source[prop] === 'object') {
-        Object.assign(target[prop], source[prop]);
-    } else {
-        target[prop] = source[prop];
+function deepCopyProperties(target, source, propertyObj) {
```

Comment:
What about `propertyObj = defaultStyle`? Then you don't need to define it above in those two places.

Reply:
If you do that, then when you create a new TextStyle with nothing passed in, that null comes through to this function and the es6 default parameter kicks in, which you don't want.

Reply:
👍